src/containers/DailyReportPage/DailyReportContainer.js

```diff
@@ -15,93 +14,152 @@ import {
   IconBtn
 } from 'components/Shared/Reports/styles';
 import Spinner from 'components/Spinner';
+import gql from 'graphql-tag';
+import { Query, graphql, compose } from 'react-apollo';
 import history from '../../utils/history';
 import { ContentWrapper } from '../../styles/App';
-import { headerItems } from './constants';
+import { headerItems, itemsAmount } from './constants';
 import formatDate from '../../utils/formatDate';
 
 const DAILY_REPORTS_QUERY = gql`
-  query DAILY_REPORTS_QUERY {
-    userReports {
-      id
-      emotion
-      title
-      achievement
-      plan
-      comment
-      createdAt
+  query DAILY_REPORTS_QUERY($first: Int, $skip: Int) {
+    userReports(first: $first, skip: $skip) {
+      count
+      dailyReports {
+        id
+        emotion
+        title
+        achievement
+        plan
+        comment
+        createdAt
+      }
     }
   }
 `;
 
-const DailyReportContainer = ({ match }) => (
-  <Query query={DAILY_REPORTS_QUERY}>
-    {({ data, error, loading }) => {
+export const USER_DAILY_REPORTS_COUNT_QUERY = gql`
+  query USER_DAILY_REPORTS_COUNT_QUERY {
+    userDailyReportsCount @client {
+      count
+      isInitialized
+      __typename
+    }
+  }
+`;
 
-      if (loading) return <Spinner />;
+export const UPDATE_USER_DAILY_REPORTS_COUNT_MUTATION = gql`
+  mutation UPDATE_USER_DAILY_REPORTS_COUNT_MUTATION($count: String!) {
+    updateUserDaiyReportsCount(count: $count) @client {
+      count
+      isInitialized
+      __typename
+    }
+  }
+`;
 
-      if (error) return <p>Error: {error.message}</p>;
+class DailyReportContainer extends Component {
+  nextPage = data => {
+    const currentPage = Number(queryString.parse(this.props.location.search).page) || 1;
+    const { count } = data.userReports;
+    const nextPage = currentPage + 1;
+    if (currentPage <= count / itemsAmount) {
+      this.props.history.push(`/reports?page=${nextPage}`);
```

Comment:
I have created a custom history in utils. 
Import `history` from `utils` and we can use this to navigate.